python/ray/tune/ray_trial_executor.py

```diff
@@ -600,6 +587,9 @@ def start_trial(self,
 
     def _find_item(self, dictionary, item):
         out = [rid for rid, t in dictionary.items() if t is item]
+        assert len(
+            out
+        ) <= 1, "Expecting one future for any given trial at any given time."
```

Comment:
Possibly unrelated to this PR: Should we rename this into `find_training_future` as we only use this helper to find futures in `self._running`?

Reply:
Yes, and instead of retuning a list, we should just return the item.
May do that separately tho. Want to first enforce this basic behavior first.